src/nemos/basis/_basis.py

```diff
@@ -135,28 +133,26 @@ class Basis(Base, abc.ABC, BasisTransformerMixin):
 
     def __init__(
         self,
-        n_basis_funcs: int,
         mode: Literal["eval", "conv"] = "eval",
         label: Optional[str] = None,
     ) -> None:
-        self.n_basis_funcs = n_basis_funcs
-        self._n_input_dimensionality = 0
+        self._n_basis_funcs = getattr(self, "_n_basis_funcs", None)
+        self._n_input_dimensionality = getattr(self, "_n_input_dimensionality", 0)
 
         self._mode = mode
 
-        self._n_basis_input = None
-
-        # these parameters are going to be set at the first call of `compute_features`
-        # since we cannot know a-priori how many features may be convolved
-        self._n_output_features = None
-        self._input_shape = None
-
         if label is None:
             self._label = self.__class__.__name__
         else:
             self._label = str(label)
 
-        self.kernel_ = None
+        # specified only after inputs/input shapes are provided
+        self._n_basis_input_ = getattr(self, "_n_basis_input_", None)
+        self._input_shape_ = getattr(self, "_input_shape_", None)
```

Comment:
what's the context in which these will be set here?

Reply:
never